### PR TITLE
#141, #144, #121 - list view's child views now follow torso lifecycles

### PR DESCRIPTION
**Pull request by kentmw**
Also upgraded backbone version to 2.1.3

### CODE REVIEW

**Comment by kentmw** [comment]
NOTE: there is a performance hit with this. We are rendering child views on list view renders

**Comment by kentmw** [comment]
Perhaps we should make it so sorting events don't cause a re-renders on the child views before we merge this.

File: modules/ListView.js [hunk 1]
```diff
@@ -319,15 +324,23 @@
...
      * @return a DOM fragment with child view elements appended
-     * @method __buildChildViewsFragment
+     * @method __emptyAndRebuildChildViewsFragment
      * @private
      */
-    __buildChildViewsFragment: function(renderAlso) {
+    __emptyAndRebuildChildViewsFragment: function(renderAlso) {
       var injectionFragment = document.createDocumentFragment();
+      // Clearing the DOM will reduce the repaints needed as we detach each child view.
+      this.$el.empty();
```
Comment (mandragorn) [inline modules/ListView.js]: this feels like it should live outside of this method.  It is mutating the parent, where this method is about generating the child content.

File: modules/ListView.js [hunk 2]
```diff
@@ -319,15 +324,23 @@
...
      _.each(this.modelsToRender(), function(model) {
         var childView = this.getChildViewFromModel(model);
         if (childView) {
+          childView.detach();
```
Comment (mandragorn) [inline modules/ListView.js]: oh, right, this mutates the parent.  nvm.

**Comment by mandragorn** [comment]
+2

**Pull request merged**